client/src/utils/getUser.ts

```diff
@@ -0,0 +1,21 @@
+import { legacyEventIds } from "../constants/legacyEventIds";
+import type { User } from "../types/User";
+
+export function getUser(eventId: string): User | undefined {
+  if (eventId in legacyEventIds) {
+    const cookieMatch = document.cookie.match(/user_details=([^;]+)/);
+    if (cookieMatch) {
+      try {
+        return JSON.parse(decodeURIComponent(cookieMatch[1]));
+      } catch (err) {
+        console.error(err);
+      }
+    }
+  } else {
+    const userDetails = JSON.parse(localStorage.getItem("my_events") ?? "{}");
```

Comment:
```suggestion
    const userDetails = JSON.parse(localStorage.getItem("my_events")) || {};
```

This probably also covers cases where `JSON.parse` returns `undefined` (gotta love JS)